src/Integration/LocalServices/ISourceControlledFileSystem.cs

```diff
@@ -35,11 +36,21 @@ internal interface ISourceControlledFileSystem : ILocalService
         /// <param name="fileWriteOperation">The actual write operation, return false if failed</param>
         void QueueFileWrite(string filePath, Func<bool> fileWriteOperation);
 
+        /// <summary>
+        /// Queues a write operation for given file paths. New or edit is determined internally at the time of this method execution.
+        /// </summary>
+        void QueueFileWrites(IList<string> filePaths, Func<bool> fileWriteOperation);
+
         /// <summary>
         /// Returns whether the file exists on disk or in a queue to be written. <seealso cref="QueueFileWrite(string, Func{bool})"/>
         /// </summary>
         bool FileExistOrQueuedToBeWritten(string filePath);
 
+        /// <summary>
+        /// Returns whether all given files exist on disk or in a queue to be written. <seealso cref="QueueFileWrites(IList{string}, Func{bool})"/>
+        /// </summary>
+        bool FilesExistOrQueuedToBeWritten(IList<string> filePaths);
+
```

Comment:
Are both `FileExist...` and `FilesExist...` required? Same for `QueueFileWrite` and `QueueFileWrite`.